var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -196,18 +199,28 @@ def filter_config_dot_pm(self):
         config_dot_pm = perl('-MModule::Loaded', '-MConfig', '-e',
                              'print is_loaded(Config)', output=str)
 
-        match = 'cc *=>.*'
-        substitute = "cc => '{cc}',".format(cc=self.compiler.cc)
-        filter_file(match, substitute, config_dot_pm, **kwargs)
+        with self.make_briefly_writable(config_dot_pm):
+            match = 'cc *=>.*'
+            substitute = "cc => '{cc}',".format(cc=self.compiler.cc)
+            filter_file(match, substitute, config_dot_pm, **kwargs)
 
         # And the path Config_heavy.pl
         d = os.path.dirname(config_dot_pm)
         config_heavy = join_path(d, 'Config_heavy.pl')
 
-        match = '^cc=.*'
-        substitute = "cc='{cc}'".format(cc=self.compiler.cc)
-        filter_file(match, substitute, config_heavy, **kwargs)
-
-        match = '^ld=.*'
-        substitute = "ld='{ld}'".format(ld=self.compiler.cc)
-        filter_file(match, substitute, config_heavy, **kwargs)
+        with self.make_briefly_writable(config_heavy):
+            match = '^cc=.*'
+            substitute = "cc='{cc}'".format(cc=self.compiler.cc)
+            filter_file(match, substitute, config_heavy, **kwargs)
+
+            match = '^ld=.*'
+            substitute = "ld='{ld}'".format(ld=self.compiler.cc)
+            filter_file(match, substitute, config_heavy, **kwargs)
+
+    @contextmanager
+    def make_briefly_writable(self, path):
+        """Temporarily make a file writable, then reset"""
+        perm = os.stat(path).st_mode
+        os.chmod(path, perm | 0o222)
```

Comment:
Shouldn't we only make this user-writable? There's no case where another user should be writing to this file in the brief time that it's writable.

Reply:
Yep, that makes sense.  Fixed in 797a659.